cli/src/main/java/software/amazon/smithy/java/cli/SmithyCall.java

```diff
@@ -0,0 +1,253 @@
+import software.amazon.smithy.java.json.JsonCodec;
+import software.amazon.smithy.model.Model;
+import software.amazon.smithy.model.shapes.ShapeId;
+
+@Command(name = "smithy-call", mixinStandardHelpOptions = true, version = "1.0",
+        description = "Smithy Java CLI")
+public class SmithyCall implements Callable<Integer> {
+    private static final JsonCodec CODEC = JsonCodec.builder().build();
+
+    private static final String AWS_JSON = "awsjson";
+    private static final String RPC_V2_CBOR = "rpcv2-cbor";
+    private static final String REST_JSON = "restjson";
+    private static final String REST_XML = "restxml";
+
+    private static final String[] BASE_RESOURCE_FILES = {
+            "aws.api.smithy",
+            "aws.auth.smithy",
+            "aws.customizations.smithy",
+            "aws.protocols.smithy"
+    };
+
+    @Parameters(index = "0", description = "Service Name")
+    private String service;
+
+    @Parameters(index = "1", description = "Name of the operation to perform on the service", arity = "0..1")
+    private String operation;
+
+    @Option(names = { "-m", "--model-path" }, description = "Path to the .smithy service model file", required = true)
+    private String modelPath;
+
+    @Option(names = "--input-path", description = "Path to a JSON file containing input parameters for the operation")
+    private String inputPath;
+
+    @Option(names = "--input-json", description = "JSON string containing input parameters for the operation")
+    private String input;
+
+    @Option(names = "--url", description = "Endpoint URL for the service")
+    private String url;
+
+    @Option(names = { "-p", "--protocol" }, description = "Communication protocol to use (options: awsjson, rpcv2-cbor, restjson, and restxml)")
+    private String protocol;
+
+    @Option(names = { "-a", "--auth" }, description = "Authentication method to use")
+    private String authType;
+
+    @Option(names = { "--aws-region" }, description = "AWS region for SigV4 authentication")
+    private String awsRegion;
+
+
+    @Option(names = "--list-operations", description = "List all available operations for the specified service")
+    private boolean listOperations;
+
+    @Override
+    public Integer call() {
+        try {
+            if (!listOperations && operation == null) {
+                throw new IllegalArgumentException("Operation is required when not listing operations");
+            }
+            return listOperations ? listOperationsForService() : executeOperation();
+        } catch (IllegalArgumentException e) {
+            logError("Invalid input", e);
+            return 1;
+        } catch (Exception e) {
+            logError("Unexpected error occurred: ", e);
+            return 1;
+        }
+    }
+
+    private Integer listOperationsForService() {
+        try {
+            Model model = assembleModel(modelPath);
+            System.out.println("Available Operations:\n" + model.getOperationShapes());
```

Comment:
You could return each operation name separated by newlines instead. That's human readable, and still easily parsed and iterated over. Then skip the heading you're printing out.